src/TreeHouse/IoBundle/Item/Modifier/Item/Filter/ModifiedItemFilter.php

```diff
@@ -45,10 +46,11 @@ public function filter(ParameterBag $item)
 
         // first try modification date
         if (null !== $mutationDate = $item->getDatetimeModified()) {
-            if ($source->getDatetimeModified() > $mutationDate) {
+            if ($source->getDatetimeImported() > $mutationDate) {
                 throw new FilterException('Item is not modified');
             }
         }
+        $source->setDatetimeImported(new DateTime());
```

Comment:
You're mutating the source state here. imho a filter should not have any side effects

Reply:
I'll have another look for a better place.